components/script/dom/imagedata.rs

```diff
@@ -55,31 +55,37 @@ impl ImageData {
             rooted!(in (*cx) let mut js_object = ptr::null_mut::<JSObject>());
             if let Some(ref mut d) = data {
                 d.resize(len as usize, 0);
+
+                let typed_array =
+                    create_buffer_source::<ClampedU8>(cx, &d[..], js_object.handle_mut(), can_gc)
+                        .map_err(|_| Error::JSFailed)?;
+
                 let data = CreateWith::Slice(&d[..]);
                 Uint8ClampedArray::create(*cx, data, js_object.handle_mut()).unwrap();
-                Self::new_with_jsobject(global, None, width, Some(height), js_object.get(), can_gc)
+                Self::new_with_data(
+                    global,
+                    None,
+                    width,
+                    Some(height),
+                    CustomAutoRooterGuard::new(*cx, &mut CustomAutoRooter::new(typed_array)),
```

Comment:
I think you want to assign this to a variable and then call the function. May be the source of the crash actually.